File: app/src/main/java/com/Fachhochschulebib/fhb/pruefungsplaner/RequestInterface.java
//////////////////////////////
// Requestinterface
//
//
//
// autor:
// inhalt: Interface damit Prüfplanobjekte von der Json Modelklasse zur lokalen Datenbank hinzugefügt werden können
// zugriffsdatum: 11.12.1992
//
//
//
//
//
//
//////////////////////////////

package com.Fachhochschulebib.fhb.pruefungsplaner;

import java.util.List;

import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonResponse;
import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonStudiengang;
import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonUpdate;
import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonUuid;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;


public interface RequestInterface {
    @GET(" ")
    Call<List<JsonResponse>> getJSON();

    //Start Merlin Gürtler
    @GET(" ")
    Call<JsonUuid> getJsonUuid();

    @PUT(" ")
    Call<Void> sendUuid();

    @POST(" ")
    Call<Void> sendFeedBack();

    @GET(" ")
    Call<List<JsonUpdate>> getJSONUpdate();

    @GET(" ")
    Call<List<JsonStudiengang>> getStudiengaenge();

    @POST(" ")
    Call<Void> sendCourses();
    //Ende Merlin Gürtler
}

File: app/src/main/java/com/Fachhochschulebib/fhb/pruefungsplaner/RequestInterface.kt
//////////////////////////////
// Requestinterface
//
//
//
// autor:
// inhalt: Interface damit Prüfplanobjekte von der Json Modelklasse zur lokalen Datenbank hinzugefügt werden können
// zugriffsdatum: 11.12.1992
//
//
//
//
//
//
//////////////////////////////
package com.Fachhochschulebib.fhb.pruefungsplaner

import retrofit2.http.GET
import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonResponse
import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonUuid
import retrofit2.http.PUT
import retrofit2.http.POST
import com.Fachhochschulebib.fhb.pruefungsplaner.model.JsonCourse
import retrofit2.Call

interface RequestInterface {
    @get:GET(" ")
    val jSON: Call<List<JsonResponse?>?>?

    //Start Merlin Gürtler
    @get:GET(" ")
    val jsonUuid: Call<JsonUuid?>?

    @PUT(" ")
    fun anotherStart(): Call<Void?>?

    @POST(" ")
    fun sendFeedBack(): Call<Void?>?

    @get:GET(" ")
    val studiengaenge: Call<List<JsonCourse?>?>?

    @POST(" ")
    fun sendCourses(): Call<Void?>? //Ende Merlin Gürtler
}